components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/LocalEntryResource.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.micro.integrator.management.apis;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.config.Entry;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Resource for local entry management in api
+ * */
+public class LocalEntryResource implements MiApiResource {
+
+    private static final Log LOG = LogFactory.getLog(LocalEntryResource.class);
+
+    // HTTP method types supported by the resource
+    Set<String> methods;
+    // Constants for local-entry json object
+    private static final String TYPE_ATTRIBUTE= "type";
+    private static final String VALUE_ATTRIBUTE= "value";
+    private static final String DESCRIPTION_ATTRIBUTE= "description";
+
+    public LocalEntryResource() {
+        methods = new HashSet<>();
```

Comment:
Onle the line below is adding an entry to the set, methods. Is it better if we initialize the HashSet with a capacity of 1?

Reply:
This is a dynamic data structure. Would it create a noticeable performance impact by setting a size?

Reply:
Yes, still it'll be initialized with a capacity of 16. Noticeable or not, we don't need to introduce any performance hit. WDYT?